Clean up packaging system

IceWizard7 opened the pull request:
This pull request cleans up and modernizes the packaging configuration:
- Migrates `requirements.txt`, `setup.py`, `setup.cfg`, `MANIFEST.in`, and `tox.ini` into `pyproject.toml`
- Removes legacy/duplicated configuration
- Updates file inclusion rules to match existing MANIFEST behavior
- No changes to runtime behavior or API

This is a build/packaging-only cleanup.

IceWizard7 committed:
Clean up packaging system

Signed-off-by: IceWizard7 <mrrioes.dc@gmail.com>

IceWizard7 commented:
Closes #67

untitaker commented:
I am in favor of this, but I propose that you manually create the sdist and wheels, extract them, and compare their contents against the same procedure done on main branch. My past experience with Python packaging tells me that this is really the only way to figure out whether two setups are equivalent.

IceWizard7 committed:
Fixed pyproject.toml; Bring back tox.ini

Signed-off-by: IceWizard7 <mrrioes.dc@gmail.com>

IceWizard7 commented:
Thanks for noticing! I absolutely agree with the need for a comparison.

I ran that exact `diff` between the build artifacts from the current main branch and my proposed changes. I just fixed another bug, my changes are now correctly implemented.

I also bumped the required Python version from >= 3.4 to >=3.8 and removed py27 from `tox`. (Those versions reached End of Life ages ago; https://devguide.python.org/versions/)
The built output is now exactly the same across both environments.

AugustH commented:
if we remove 2.7 support we should refactor a lot of code with old language constructs. 

But i think there are a lot of old, long time used scripts out there (running on 2.7 or 3.x, x < 8) which would be excluded from any changes -- at least i have some old things "running".

IceWizard7 commented:
Python2.7 should really never be used, as it stopped getting security updates about 6 years ago.
To clarify, only the [`tox.ini` file lists Python2.7](https://github.com/python-webuntis/python-webuntis/blob/master/tox.ini#L3) on the main branch right now, but [setup.py doesn't](https://github.com/python-webuntis/python-webuntis/blob/master/setup.py#L34), the oldest version there is 3.4.

I suppose we could change both of these files to Py3.4 to keep it consistent...

IceWizard7 commented:
Also, I just noticed that the [read the docs](https://python-webuntis.readthedocs.io/en/latest/) page says >=Py2.6 & >=Py3.3... I'm not saying we need to remove support for legacy versions, but we should at least make this consistent.

IceWizard7 commented:
Any more questions or proposals? @untitaker @AugustH 
Or is this ready to merge?

IceWizard7 commented:
Will this get merged? @untitaker @maphy-psd @AugustH

untitaker commented:
I don't have the bandwidth to review that right now

IceWizard7 commented:
More stable Python API [here](https://github.com/IceWizard7/untis).
More stable C++ API [here](https://github.com/IceWizard7/untis-cpp).

Not commenting for 4 months on this just proves how unstable this project is.
